pkgs/development/python-modules/serverlessrepo/default.nix

```diff
@@ -1,13 +1,24 @@
 { lib
-, buildPythonPackage
-, fetchPypi
-, pytest
-, boto3
-, six
-, pyyaml
-, mock
+, python
 }:
 
+let
+  py = python.override {
+    packageOverrides = self: super: {
+      pyyaml = super.pyyaml.overridePythonAttrs (oldAttrs: rec {
```

Comment:
overriding a Python library inside `python-packages.nix` is going to cause trouble when composing an env consisting of the original version and the overridden version

Reply:
@FRidh understood. I'll get a fix in ASAP.

serverlessrepo has a dependency on `pyyaml` declared as ` 'pyyaml~=3.12'`.

I'm not much of a python developer, and I wonder what's the best way to deal with this? Should I fix up version in `setup.py`, or package an older version of `pyyaml`?

Reply:
a) check (yourself/with upstream) whether it actually does work with the version we use, and if so patch it
b) if it does not work, then the module is unfortunately broken. Even so, an application can choose to perform the override as is done here

Reply:
Thanks @Fridh. The tests are passing for python 2.7 after patching. Will make another PR after the tests also pass for 3.5, 3.6 and 3.7.

Apologies for the trouble.

Reply:
Ouch, I actually knew about this, but totally missed that the change is done to a library, sorry! :/

@dhl do you need assistance to fix that?

Reply:
@Ma27 I've fixed this in https://github.com/NixOS/nixpkgs/pull/62051